### PR TITLE
feat: dev only verifier templates

### LINKED ISSUE #1213: Review/Update proof request templates descriptions
Review and update the description for all proof request templates

https://github.com/bcgov/bc-wallet-mobile/blob/384077e2ce14cbb4dabed67783e55b88e65b020e/app/src/request-templates.ts#LL54C53-L54C53

### PR DESCRIPTION
**Pull request by wadeking98**
Sheldon asked for Unverified Person templates for ease of testing, Decided to make them development only templates so they don't end up in prod.
- Added extra toggle on developer menu to include development templates
- Bound inputs so that when the dev templates toggle is on, verifier feature must be enabled. Conversely when the verifier feature is turned off, dev template must also be turned off 

Resolves: #1213 

https://github.com/hyperledger/aries-mobile-agent-react-native/assets/36937407/d2910669-c842-4501-b2da-3d351cb80ed2

### CODE REVIEW

File: app/src/request-templates.ts [hunk 1]
```diff
@@ -50,41 +64,25 @@ export const proofRequestTemplates: Array<ProofRequestTemplate> = [
...
-                { schema_id: '2K2h7kf8VGTLtfoxJgWazf:2:Person:1.1', issuer_did: '2K2h7kf8VGTLtfoxJgWazf' }, // Dev & Test
-              ],
+              restrictions: personRestrictions,
             },
           ],
           requestedPredicates: [
             {
               name: 'birthdate_dateint',
-              predicateType: '>=',
+              predicateType: '<=',
```
Comment (bryce-mcmath) [inline app/src/request-templates.ts:83]: Are we sure this is a change we want? It seemed to be working before

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -187,7 +187,9 @@ jobs:
           -archivePath AriesBifold.xcarchive \
           -sdk iphoneos \
           -verbose \
-          archive
+          archive \
+          CODE_SIGNING_ALLOWED=NO \
+          CODE_SIGNING_REQUIRED=NO
```
Comment (bryce-mcmath) [inline .github/workflows/main.yaml]: Did you mean to include this? If so, should probably be a separate PR I think

**Review by bryce-mcmath — CHANGES_REQUESTED** [review]

File: app/src/request-templates.ts (hunk 1 above)
Comment (wadeking98) [reply app/src/request-templates.ts:83]: It only works on the same day the credential was issued. If the showcase person credential was issued on the previous day (July 9th) and today is July 10th then the request will fail because July 9th is less than July 10th. The reason it was set to ">=" in the first place was because when the templates were originally created, they used exact age instead of date of birth. ie: "date of birth >= 19" instead of "date of birth >= current date - 19 years" which was a bug because any date_int number would be greater that 19. When I fixed the bug I forgot to flip the direction of the ">=" sign.

File: .github/workflows/main.yaml (hunk 2 above)
Comment (wadeking98) [reply .github/workflows/main.yaml]: woops ya I don't know how that got there

**Review by bryce-mcmath — APPROVED** [review]
